ludwig/utils/fs_utils.py

```diff
@@ -33,14 +33,41 @@
 from filelock import FileLock
 from fsspec.core import split_protocol
 
+from ludwig.constants import S3
+from ludwig.utils.remote_storage_options import S3RemoteStorageOptions
 
-def get_fs_and_path(url):
+logger = logging.getLogger(__name__)
+
+
+def create_fs(protocol: Union[str, None], storage_options: Optional[Dict[str, Any]]) -> fsspec.filesystem:
+    """Create filesystem object with correct storage options based on the protocol."""
+    if not protocol:
+        # Defaults to LocalFileSystem
+        return fsspec.filesystem(protocol)
+
+    if not storage_options:
+        logger.info(f"Using default storage options for `{protocol}` filesystem.")
+        if protocol == S3:
+            s3 = S3RemoteStorageOptions()
```

Comment:
Why is s3 handled specially here? Do we need to do anything for other filesystems?

One way to make this clearer could be:

```
if not storage_options:
    logger.info(f"Using default storage options for `{protocol}` filesystem.")
    if protocol == S3:
        storage_options = S3RemoteStorageOptions().get_storage_options()

 try:
    return fsspec.filesystem(protocol, **storage_options)
...
```

Reply:
Following up on my comment below, I now believe that this `storage_options` handling shouldn't be necessary.

Also, this approach has an additional bug, in that it will break existing `use_credentials` behavior by overriding it with credentials from the environment (when both are using s3).

Reply:
@tgaddair S3 is handled especially here because users may want to pass in a different S3 compatible storage like Minio. The issue is that to do this, users need to pass in the `--endpoint_url` flag to AWSCLI because AWSCLI currently doesn't have an environment variable that can be configured to pick up this URL automatically (although they are currently working on a proposal to add it in: https://github.com/aws/aws-sdk/issues/229). Within fsspec, that would involve manually passing in client_kwargs with the endpoint_url defined in the object. I think this is the only reason to do something special for S3 at the moment. 

I can't entirely say what would be required for other file systems yet, but those should automatically get created via fsspec.

Reply:
About the `use_credentials` method - I think that this actually won't be an issue because both `use_credentials` and this approach will ned up using the same key and secret since they're derived from the same environment variables. The only thing that changes is the endpoint URL, which in either case would need to change for S3 compatible storage since use_credentials would also require this new endpoint URL (if it is specified).